src/coreclr/tools/aot/ILCompiler.LLVM/CodeGen/ILToLLVMImporter.cs

```diff
@@ -5228,6 +5242,18 @@ private void ImportFallthrough(BasicBlock next)
         private const string ThreadStatics = "ThreadStatics";
         private const string ClassConstructorRunner = "ClassConstructorRunner";
 
+        private ExpressionEntry LoadInstanceField(LLVMValueRef thisValueRef, FieldDesc fieldDesc, TypeDesc fieldTypeDesc, StackValueKind stackValueKind, LLVMBuilderRef builder = default(LLVMBuilderRef))
+        {
+            if (builder.Handle == IntPtr.Zero)
+                builder = _builder;
+
+            LLVMValueRef fieldAddress = _builder.BuildGEP(thisValueRef, new LLVMValueRef[] { LLVMValueRef.CreateConstInt(LLVMTypeRef.Int32, (ulong)fieldDesc.Offset.AsInt) });
```

Comment:
Is `thisValueRef` guaranteed to be `i8*` here s.t. GEP adds the right offset?

Reply:
Yes, I think we are ok, It is only set from one path:
```
thisRef = thisPointer.ValueAsType(LLVMTypeRef.CreatePointer(LLVMTypeRef.Int8, 0), _builder);
```